x-pack/plugins/apm/server/lib/transactions/get_top_transactions.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { get } from 'lodash';
+import {
+  PROCESSOR_EVENT,
+  SERVICE_NAME,
+  TRANSACTION_TYPE
+} from '../../../common/constants';
+import { StringMap } from '../../../typings/common';
+import { Transaction } from '../../../typings/Transaction';
+import { ITransactionGroup } from '../../../typings/TransactionGroup';
+import {
+  prepareTransactionGroups,
+  TRANSACTION_GROUP_AGGREGATES
+} from '../helpers/transaction_group_query';
+
+export async function getTopTransactions({
+  setup,
+  transactionType,
+  serviceName
+}: {
+  setup: StringMap<any>;
```

Comment:
You could use the new `Setup` type. But don't worry about it now if you get a green build. We can always fix that later.